common/src/main/java/io/quarkus/ts/openshift/common/util/ReadinessUtil.java

```diff
@@ -0,0 +1,29 @@
+package io.quarkus.ts.openshift.common.util;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.client.internal.readiness.Readiness;
+import io.fabric8.openshift.api.model.DeploymentConfig;
+import io.fabric8.openshift.client.internal.readiness.OpenShiftReadiness;
+
+/**
+ * Unifies Fabric8 Kubernetes Client's {@link Readiness} and {@link OpenShiftReadiness}.
+ * No other class should refer to them directly; using {@code ReadinessUtil} is preferred.
+ */
+public class ReadinessUtil {
+    public static boolean isReadinessApplicable(Class<? extends HasMetadata> resourceClass) {
+        // unfortunately `OpenShiftReadiness` doesn't have a `isReadinessApplicable` method
```

Comment:
design flaw ?

Reply:
More like an implementation omission. I wanted to submit a PR and then forgot, thanks for reminder :-)

Reply:
Once that in and released, this util can be simplified .. I guess